test/src/graph/nodes/TransformPointsNodeTest.cpp

```diff
@@ -0,0 +1,89 @@
+#include <RGLFields.hpp>
+#include <graph/Node.hpp>
+#include <gtest/gtest.h>
+#include <repr.hpp>
+#include <utils.hpp>
+
+class TransformPointsNodeTest : public RGLAutoCleanupTestWithParam<std::tuple<int, rgl_mat3x4f>>, public RGLPointTestHelper {
+};
+
+INSTANTIATE_TEST_SUITE_P(
+    TransformPointsNodeTests, TransformPointsNodeTest,
+    testing::Combine(
+        testing::Values(1, 10, 100000),
+        testing::Values(identityTestTransform, translationTestTransform, rotationTestTransform, scalingTestTransform, complexTestTransform)));
+
+TEST_P(TransformPointsNodeTest, invalid_arguments)
+{
+    auto transform = std::get<1>(GetParam());
+    rgl_node_t transformPointsNode = nullptr;
+
+    auto initializeArgumentsLambda = [&transform, &transformPointsNode]() {
+        transform = std::get<1>(GetParam());
+        transformPointsNode = nullptr;
+    };
+
+    initializeArgumentsLambda();
+    EXPECT_RGL_INVALID_ARGUMENT(rgl_node_rays_transform(nullptr, nullptr), "node != nullptr");
+
+    initializeArgumentsLambda();
+    EXPECT_RGL_INVALID_ARGUMENT(rgl_node_rays_transform(nullptr, &transform), "node != nullptr");
+
+    initializeArgumentsLambda();
+    EXPECT_RGL_INVALID_ARGUMENT(rgl_node_rays_transform(&transformPointsNode, nullptr), "transform != nullptr");
+}
+
+TEST_P(TransformPointsNodeTest, valid_arguments)
+{
+    auto transform = std::get<1>(GetParam());
+    rgl_node_t transformPointsNode = nullptr;
+
+    EXPECT_RGL_SUCCESS(rgl_node_rays_transform(&transformPointsNode, &transform));
+    ASSERT_THAT(transformPointsNode, testing::NotNull());
+
+    // If (*node) != nullptr
+    EXPECT_RGL_SUCCESS(rgl_node_rays_transform(&transformPointsNode, &transform));
+}
+
+TEST_P(TransformPointsNodeTest, use_case)
+{
+    auto [pointsCount, transform] = GetParam();
+    auto inPoints = GenerateTestPointsArray(pointsCount);
+
+    rgl_node_t usePointsNode = nullptr;
+    rgl_node_t transformNode = nullptr;
+
+    EXPECT_RGL_SUCCESS(rgl_node_points_from_array(&usePointsNode, inPoints.data(), inPoints.size(), pointFields.data(), pointFields.size()));
+    ASSERT_THAT(usePointsNode, testing::NotNull());
+
+    EXPECT_RGL_SUCCESS(rgl_node_points_transform(&transformNode, &transform));
+    ASSERT_THAT(transformNode, testing::NotNull());
+
+    EXPECT_RGL_SUCCESS(rgl_graph_node_add_child(usePointsNode, transformNode));
+
+    EXPECT_RGL_SUCCESS(rgl_graph_run(usePointsNode));
+
+    int32_t outCount, outSizeOf;
+    EXPECT_RGL_SUCCESS(rgl_graph_get_result_size(transformNode, XYZ_F32, &outCount, &outSizeOf));
+
+    EXPECT_EQ(outCount, inPoints.size());
+    EXPECT_EQ(outSizeOf, getFieldSize(XYZ_F32));
+
+    std::vector<::Field<XYZ_F32>::type> outData;
+    outData.reserve(outCount);
```

Comment:
`reserve` does not change size!
```suggestion
    outData.resize(outCount);
```